addons/binding/org.openhab.binding.icloud/src/main/java/org/openhab/binding/icloud/internal/Connection.java

```diff
@@ -0,0 +1,115 @@
+
+    private final byte[] authorization;
+    private URL iCloudDataRequestURL;
+    private URL iCloudPingDeviceURL;
+
+    public Connection(String appleId, String password) throws MalformedURLException {
+        authorization = Base64.getEncoder().encode((appleId + ":" + password).getBytes());
+        iCloudDataRequestURL = new URL(iCloudApiURL + appleId + iCloudAPIRequestDataCommand);
+        iCloudPingDeviceURL = new URL(iCloudApiURL + appleId + iCloudAPIPingDeviceCommand);
+    }
+
+    public String requestDeviceStatusJSON() throws Exception {
+        HttpsURLConnection connection = postRequest(iCloudDataRequestURL, dataRequest);
+
+        String response = getResponse(connection);
+        return response;
+    }
+
+    /***
+     * Sends a ping request (find my phone).
+     *
+     * @throws Exception
+     */
+    public void pingPhone(String id) throws Exception {
+        String request = "{ \n \"device\": \"" + id + "\", \n \"subject\": \"Find My Device alert\" \n }";
+        getResponse(postRequest(iCloudPingDeviceURL, request));
+    }
+
+    private void setRequestProperties(HttpsURLConnection connection, String payload)
+            throws UnsupportedEncodingException {
+        connection.setRequestProperty("Authorization", this.getBasicAuthorization());
+        connection.setRequestProperty("User-Agent", "Find iPhone/1.3 MeKit (iPad: iPhone OS/4.2.1)");
+        connection.setRequestProperty("Origin", iCloudURL);
+        connection.setRequestProperty("Content-Type", "application/json");
+        connection.setRequestProperty("charset", "utf-8");
+        connection.setRequestProperty("Accept-language", "en-us");
+        connection.setRequestProperty("Connection", "keep-alive");
+        connection.setRequestProperty("X-Apple-Find-Api-Ver", "2.0");
+        connection.setRequestProperty("X-Apple-Authscheme", "UserIdGuest");
+        connection.setRequestProperty("X-Apple-Realm-Support", "1.0");
+        connection.setRequestProperty("X-Client-Name", "iPad");
+        connection.setRequestProperty("Content-Length", Integer.toString(payload.getBytes("UTF-8").length));
+    }
+
+    private HttpsURLConnection postRequest(URL url, String payload) throws Exception {
+        HttpsURLConnection connection;
+        connection = (HttpsURLConnection) url.openConnection();
+        connection.setDoOutput(true);
+        connection.setDoInput(true);
+        connection.setRequestMethod("POST");
+        setRequestProperties(connection, payload);
+
+        connection.connect();
+        connection.getOutputStream().write(payload.getBytes("UTF-8"));
+        connection.disconnect();
+        return connection;
+    }
+
+    private String getResponse(HttpsURLConnection connection) throws IOException {
+        String response;
+        BufferedInputStream inputStream = new BufferedInputStream(connection.getInputStream());
+        BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
```

Comment:
I suppose `new BufferedReader(new InputStreamReader(connection.getInputStream()));` should do the job

Reply:
Please use a try with resources here:

```java
try(BufferedReader reader =new BufferedReader(new InputStreamReader(connection.getInputStream()))){
 ....
}
```
This handles the closing logic and guarantees to close also in case of exceptions

Reply:
Maybe the same tip applies here to use the HttpUtil